integration-tests/deployment/multiclient.go

```diff
@@ -95,30 +142,47 @@ func (mc *MultiClient) CodeAt(ctx context.Context, account common.Address, block
 
 func (mc *MultiClient) NonceAt(ctx context.Context, account common.Address) (uint64, error) {
 	var count uint64
-	err := mc.retryWithBackups(func(client *ethclient.Client) error {
+	err := mc.retryWithBackups(ctx, func(client *ethclient.Client) error {
 		var err error
 		count, err = client.NonceAt(ctx, account, nil)
 		return err
 	})
 	return count, err
 }
 
-func (mc *MultiClient) retryWithBackups(op func(*ethclient.Client) error) error {
-	var err error
-	for _, client := range append([]*ethclient.Client{mc.Client}, mc.Backups...) {
-		err2 := retry.Do(func() error {
-			err = op(client)
-			if err != nil {
-				// TODO: logger?
-				fmt.Printf("Error %v with client %v\n", err, client)
-				return err
-			}
-			return nil
-		}, retry.Attempts(mc.RetryConfig.Attempts), retry.Delay(mc.RetryConfig.Delay))
+// retryWithBackups will retry the operation with the main client and then with the backup clients
+// If the operation returns an error that is in the acceptedErrors list, it will not retry the operation or try the next client
+func (mc *MultiClient) retryWithBackups(ctx context.Context, op func(*ethclient.Client) error, acceptedErrors ...error) error {
+	var err2 error
+	funcName := runtime.FuncForPC(reflect.ValueOf(op).Pointer()).Name()
+	for i, client := range append([]*ethclient.Client{mc.Client}, mc.Backups...) {
+		err2 = retry.Do(ctx, retry.WithMaxRetries(mc.RetryConfig.Attempts, retry.NewConstant(mc.RetryConfig.Delay)),
+			func(ctx context.Context) error {
+				err := op(client)
+				if err != nil {
+					// Check if the error is one of the accepted errors
+					// If it is, log it and return nil
+					for _, acceptedError := range acceptedErrors {
+						if errors.Is(err, acceptedError) {
+							return err
```

Comment:
`// If it is, log it and return nil` did you mean to return `nil` here?

Reply:
updated the comment